### DISCUSSION TITLE
Feature Request - Network Throttling

### DISCUSSION THREAD

**Discussion by drnichols** [lavinir/hassio-onedrive-backup#70] [Ideas]
Hi, I'm currently using the Backup to Google Drive addon and one issue I've had is with upload bandwidth saturation during backup uploads to the cloud.

Would it be possible to implement upload throttling in your add-on? 

I know the windows onedrive client offers this capability.

Would be a compelling feature for me if you could implement this (along with being able to use onedrive business).

Thank you for your efforts to develop this addon.

**Comment by lavinir** [comment]
Throttling would be an interesting feature to implement :) 
Out of curiosity - limiting the backup / sync window to certain hours of the day (which is possible) is not sufficient ? Is it still a bandwith concern if you're syncing at 3:00am for example ?

As to OneDrive for Business, this is not possible until Microsoft allow using the App folder permission scope for a Work/School account otherwise I would require access to all the files and folders in OneDrive.
